daemon/api_test.go

```diff
@@ -1357,21 +1283,13 @@ func (s *apiSuite) TestLoginUserDeveloperAPIError(c *check.C) {
 
 	c.Check(rsp.Type, check.Equals, ResponseTypeError)
 	c.Check(rsp.Status, check.Equals, 401)
-	c.Check(rsp.Result.(*errorResult).Message, testutil.Contains, "cannot get snap access permission")
+	c.Check(rsp.Result.(*errorResult).Message, testutil.Contains, "error-from-login-user")
 }
 
 func (s *apiSuite) TestLoginUserTwoFactorRequiredError(c *check.C) {
-	serializedMacaroon, err := s.makeStoreMacaroon()
-	c.Assert(err, check.IsNil)
-	responseData, err := s.makeStoreMacaroonResponse(serializedMacaroon)
-	c.Assert(err, check.IsNil)
-	mockDeveloperAPIServer := s.makeDeveloperAPIServer(200, responseData)
-	defer mockDeveloperAPIServer.Close()
-
-	discharge := `{"code": "TWOFACTOR_REQUIRED"}`
```

Comment:
The low level bits are tested in auth_test.go:TestDischargeAuthCaveatNeeds2fa().